### ISSUE TITLE
Unified player should be optional, makes the app useless in certain scenarios

### ISSUE THREAD

**Issue by nbmrjuhneibkr** [TeamNewPipe/NewPipe#4460]
### Checklist

- [x] I am using the latest version.
- [x] I checked, but didn't find any duplicates of this issue in the repo.
- [x] I have read the contribution guidelines given at https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md.
- [x] This issue contains only one bug. I will open one issue for every bug report I want to file.

### Steps to reproduce the bug

New "unified player" makes the app useless if I want to use background player independently from the video player. I'm filing this as a bug, because this is something that worked perfectly fine in v0.18, but is completely broken now due to some questionable features being implemented in v0.20.  

### Actual behaviour
When I have tracks in the background queue (either playing or paused), and then try to watch a video in NewPipe, my background queue (which is valuable user data) gets erased without any warnings. This is default behavior in 0.20. The option "Ask for confirmation before clearing a queue" (which is disabled by default) adds a warning (that sometimes doesn't work -  https://github.com/TeamNewPipe/NewPipe/issues/4459 ), but does nothing to solve the actual problem. The fact that this option even needs to exist now is a clear indication that current implementation is wrong.

I have music and podcasts in my background queue pretty much 24/7, which means that I can't use NewPipe as a video player anymore without constantly losing my queue. And combined with https://github.com/TeamNewPipe/NewPipe/issues/4458 this means that now I can't even open a video from NewPipe in another player without losing my background queue.

### Expected behavior
There should be an option to use background player independently from the video player, just like in previous versions. These players serve a different purpose, and may be used for very different types of content.

### Screenshots/Screen recordings
0.18:
![018a](https://user-images.githubusercontent.com/8460563/95567665-f4b05380-0a2b-11eb-90b3-85f44021da7f.png)
![018b](https://user-images.githubusercontent.com/8460563/95567681-f843da80-0a2b-11eb-9919-7fdfb1058098.png)

0.20:
![020a](https://user-images.githubusercontent.com/8460563/95567721-04c83300-0a2c-11eb-9e56-c10bc842c416.png)
![020b](https://user-images.githubusercontent.com/8460563/95567730-0691f680-0a2c-11eb-80f9-0e702730f946.png)
![020c](https://user-images.githubusercontent.com/8460563/95567733-07c32380-0a2c-11eb-91c6-50002f35c42a.png)

### Device info

 - Android version/Custom ROM version: Android 11
 - Device model: Pixel 3a

**Comment by hipunk** [comment]
Yes! Pleeaase make it optional!

I can only speak for myself, but let me tell you this:
I'm almost always listening to something using newpipe with the background player. Podcasts and music, it happens very often.
In 0.19 and earlier: if somebody sent me a youtube link in e.g. WhatsApp or Discord or whatever, I'd tap it and open it with newpipe using its popout player (most of the time, that is).
Now, with 0.20.0 I can't do that anymore, because if I'd do that, the music or podcast wouldn't only pause playing, it'd even stop & vanish from the queue. So now I oftentimes choose to open the youtube link with my browser, e.g. Chrome, in order to prevent the music or podcasts from getting lost. You devs should really be aware of this.

Do you, as a dev, really want your users to use your app less than before?

**Comment by opusforlife2** [comment]
Hmmm... option to have separate queues for the 3 players? Maybe pick and choose 2 players to have a common queue (like main and popup, but separate background queue)?

**Comment by nbmrjuhneibkr** [comment]
I don't use the popup player, so I can't comment on that, but there definitely needs to be an option to have a separate queue for the background player. Lots of people keep their audio and video players separate from each other.

**Comment by opusforlife2** [comment]
I would love that option as well. What you play in the background player is usually very different from what you play in the main or popup players.

**Comment by avently** [comment]
@nbmrjuhneibkr 
> New "unified player" makes the app useless

>but is completely broken

> questionable features

So you think that when you talk like this the devs will want to help you?

**Comment by nbmrjuhneibkr** [comment]
I'm thinking that it's rather strange that the devs let a major regression like this make its way into release after such a long development process.

All of the statements you quoted are factually correct in the context of this issue.

**Comment by opusforlife2** [comment]
The problem is precisely that "long development process" you mention. Because this was such a massive change, there were bound to be lots of bugs in the final release, even though we tried to spot as many of them as possible. Now what we need is rapid user feedback and some small releases to address critical bugs, and eventually those of lesser and lesser urgency in further releases.

**Comment by nbmrjuhneibkr** [comment]
The problem with this one is that it isn't really a bug, but rather a deliberate design decision that completely removes previously possible usage scenarios due to its limitations. I submitted it as a bug anyway, and described my reasons for doing this.

**Comment by opusforlife2** [comment]
I wasn't addressing this bug in particular, but just the general flood of new issues opened after the release was published. We expected this would happen. Now we will work to address them one by one.

**Comment by panacist** [comment]
for the seek of playing 360p videos on my replicant os i do give my vote on this! unified player should be optional.

**Comment by kapodamy** [comment]
seconding for making it optional, there so many bugs (or weird behaviors) thats make imposible to use

**Comment by opusforlife2** [comment]
Oh hey, long time no see, @kapodamy! Why are you not on the IRC channel?

**Comment by streaps** [comment]
I would agree, these are not code bugs, but usability bugs = bad UI design and bad UX due to sudden changes in the behaviour. The new version is kind of depressing (UX wise. It seems the app is more stable, crashing less often and a bit snappier).

**Comment by yiiii777** [comment]
> So you think that when you talk like this the devs will want to help you?

Maybe devs wants new fork of entire project?

For me, version 0.20 is totally unusable for me, I was very angry and downgraded to 0.18... But when YouTube will change their interface, what options remains for me? Maybe I can find any other player? Or write another, "very tiny and simple, without bloatware" app? Like newpipe was sometime...

**Comment by yiiii777** [comment]
> crashing less often and a bit snappier

Got in few minutes of usage
![Screenshot_20201011-053246_NewPipe](https://user-images.githubusercontent.com/69321035/95669020-43074480-0b49-11eb-9f53-bab26c845066.png)

**Comment by Stypox** [comment]
> Got in few minutes of usage

Will be fixed in 0.20.1

**Comment by hp8wvvvgnj6asjm7** [comment]
the latest release was a total gnome move. Can someone please fork this project before it gets out of hand?

**Comment by opusforlife2** [comment]
> the latest release was a total gnome move. Can someone please fork this project before it gets out of hand?

You're a Github user. Be proactive and use the Fork button. :P

**Comment by hp8wvvvgnj6asjm7** [comment]
@opusforlife2 is this a nice way of saying, go fork yourself? :D

**Comment by opusforlife2** [comment]
That is genuinely hilarious. I bow to your comedic genius. xD

**Comment by kubo6472** [comment]
ClassicPress 2.0?

**Comment by cyber199** [comment]
While I don't disagree with the unified player, I can see the background queue being erased as being a problem.  I could see a user trying to enqueue a YouTube video or Soundcloud song in the background while listening to their queue and accidentally erasing the list.  The background player, i.e. the one with no video, would benefit from not being unified with the foreground players (fullscreen and popup) in this scenario.

**Comment by yashpalgoyal1304** [comment]
I agree this should have been made as a separate release - as beta program of some such.

The problem is - i cant even downgrade as i dont want to lose my history and fqin android v7.0 wont let me install an older app version *over* current one.

So, can 0.19.x be released as 0.20.1 or smth so that i can update it. Internally it can still be referred to as 0.19.... i dont know... help!

**Comment by Moon-Palace-16** [comment]
I have to agree that the new version should probably have been a separate version of the app. I'm not really fond of some of the changes, some of which don't even work with my operating system (like the new notification bar, which no longer shows how the long the video is, or progress, and is stuck with a medium gray color). I would love it if someone would fork the older version, and maybe make two separate versions of the app. I know that's a lot of work, though, and I don't mind changes if everything still works well, but I feel like a lot of new issues have appeared instead with the last two updates.

I did switch back to 0.19.8, but that is now starting to give me some issues with playing videos in the background, so I'll soon be forced to update, but there are a few issues which have been turning me away a bit.

This said, I do want you guys to know that I absolutely love your app, and use it almost every single day (mainly for music), usually for hours.

**Comment by Flashwalker** [comment]
Nowaday's realities bombard us with tones of info channels. So it was very convinient to use background player for listening, floating player for watching while reading/browsing, and full player for fullscreen, info and comments and etc.

**Comment by Flashwalker** [comment]
Well... when you have two options and choose the first to push into the background and getting the second one... it looks like there is a risk of misunderstanding in the operation of the application when you wait for one and get another

**Comment by rancidfrog** [comment]
Unified player has bugs.
Double queued videos in some cases,
Switching to main player causes video to start playing even though it is not in background player queue.
It seems it has not been tested properly for all use-cases.
Thorough testing needed for all edge cases, too.

**Comment by Flashwalker** [comment]
Unified player and some recent changes significantly degraded the once good application

**Comment by yashpalgoyal1304** [comment]
> Unified player and some recent changes significantly degraded the once good application

@flashwalker what are the changes other than unified player 🤔 that “degrade” it??

**Comment by Flashwalker** [comment]
@yashpalgoyal1304 for example, try "open with" a video that newpipe can'ot load.
You won't even be able to access this button

**Comment by SameenAhnaf** [comment]
I think, handling too many queues at the same time will be a little difficult as I might forget what the queues are about if media player notification bar is not shown. If shown, having several queues on is not a good idea as the notification panel will get bombarded with too many media player notification bars.

 How about if we add "Add to playlist" in "Always ask" share menu mentioned in #5858?

After adding to playlist, we can watch similar videos later. This will also avoid accidental removal of videos from enqueued lists.

If a user still wants to watch popup right after background and then they can add all videos and use "Always-on button" mentioned in #5821. It will be a 2 in 1, users will be able to switch to background or popup whenever within a single tap. No need to even pull down notification panel.

**Comment by Stypox** [comment]
I like the "add to playlist" button

**Comment by nbmrjuhneibkr** [comment]
Originally this issue was called "Unified player should be optional", not "Option to have separate queues". There is no need to have multiple queues for the usage scenario described in this issue. In v0.19 the background audio queue was totally unaffected by the video playback, and videos were never added to any queue when you simply watched them in NewPipe. If this behavior was available in 0.20 (as an option) instead of being completely removed, there would've been no need to come up with convoluted workarounds like the one described in https://github.com/TeamNewPipe/NewPipe/issues/4460#issuecomment-803043376 (that doesn't really solve the actual problem described here).

**Comment by Flashwalker** [comment]
> Originally this issue was called "Unified player should be optional", not "Option to have separate queues".  
Yep, you are right

**Comment by SameenAhnaf** [comment]
@nbmrjuhneibkr Extremely sorry, but you missed a point. Even if you don't mess up the queues, you will end up having multiple media player notification bars which is certainly unnecessary.

Most expectedly, similar videos are enqeued. Adding them to a playlist and watching them should not be a hassle.

Even, if you want to add multiple types of videos (for example, watch an animated motivational video, then a podcast), add them in a single queue and switch using "Always-on button" in #5821. Here, you will be able to assign "Switch to background" and "Switch to popup". No need to drag down notification panel. Plus, you will be able to switch within a single tap.

**Comment by nbmrjuhneibkr** [comment]
This issue was never about "watching them later", it's about watching them _immediately_, without any changes to existing background queue.
There is no need to have _any_ media notifications for a single video that is being played in NewPipe.
All of this was already implemented and worked just fine in v0.19.

**Comment by SameenAhnaf** [comment]
@nbmrjuhneibkr Why do you need to manage multiple queues at the same time? If you like to switch very often, I suggested an option for that too.

>Even, if you want to add multiple types of videos (for example, watch an animated motivational video, then a podcast), add them in a single queue and switch using "Always-on button" in #5821. Here, you will be able to assign "Switch to background" and "Switch to popup". No need to drag down notification panel. Plus, you will be able to switch within a single tap.

**Comment by nbmrjuhneibkr** [comment]
> Why do you need to manage multiple queues at the same time?

**I don't.** Did you even read my comments? I don't need video to be added to _any_ queue when I simply press the Play button and watch it. I don't need any media notifications while it's playing either.
This is how it already worked in v0.19 and earlier. Have you ever used NewPipe before v0.20?

This issue is not a feature request, because the feature already existed in NewPipe, and we know exactly how it's supposed to work.

**Comment by nbmrjuhneibkr** [comment]
I'm sorry, @opusforlife2 , but I have to change the title of this issue back to the original one. It seems like new title creates more confusion and leads to people suggesting  workarounds that are unrelated to the actual issue.

**Comment by SameenAhnaf** [comment]
@nbmrjuhneibkr I completely understand your suggestion. But there are multiple things we can do.

1. Reverting back (As you suggested): We will have multiple queues back. But the problem is that we won't be able to manage the queues we are not playing. The queue will have to by opening another media which is certainly not a good thing as you may not need any other media.

2. Have multiple notifications for each queue: That's neither a good idea as the notifications will require a lot of space and more necessary notifications might get hidden.

2. Have a single queue and switch using an "Always-on button" and switch whenever you want (as I suggested): #5821 explains how to add switching options. It won't require much hassle as you can manage the queue from notification and switch to any other player.

**Comment by nbmrjuhneibkr** [comment]
@SameenAhnaf Enough. I have explained multiple times why this issue has nothing to do "multiple queues" or media notifications.
Please stop suggesting your own feature request https://github.com/TeamNewPipe/NewPipe/issues/5821 as a solution for unrelated issues.

**Comment by avently** [comment]
>There is no need to have any media notifications for a single video that is being played in NewPipe.


It's not true.


> I don't need any media notifications while it's playing either

You don't know how all the code works. It's not about what you want it's about how to make the player working. Notification is the only solution for playing something in android in a service.

>This is how it already worked in v0.19 and earlier. Have you ever used NewPipe before v0.20?

It's not true too. Did you ever used NewPipe 0.19 before? When popup and background players work both notifications appear. Two notifications, not one. In case of three players there should be three notifications at a time. Did you understand or not? If not, ask your questions. Maybe by understanding the underlying changes you'll be able to find a solution for your problem

**Comment by avently** [comment]
I would say that it is possible to have separate queues but one service/notification. But someone needs to find a way to control all three queues in a comfortable way. That's hard. Let's say you're playing something in main player. Then you want to play something in background player. Main player pauses but UI is visible (play button and others, video is shown instead of thumbnail). Notification changes it's content to background queue instead of main player.

Then you choose to play something in popup. Same as with previous example. But I don't know how to access background queue. probably via additional button in mini player at the bottom. So one button in mini player switches to background queue after pressing. Popup and main player still shows a picture of the paused video and it's controls but sound plays from backgroud queue. And click on notification will guide you to background queue.

Whenever you play something within other player like popup or main, queue switches to it as well as notification. 

For this to work you need to have two cached player's UI for popup and main so they can coexist. 
And you need additional button in mini player when background queue is exist. 
And you need to have three queue available at a time. The switching between them should be an easy thing to do. 
And you have to return back buttons like "Play in popup", "play in background". 

Actually all this idea is not hard to implement. Of course this behavior should be optional.

**Comment by SameenAhnaf** [comment]
@nbmrjuhneibkr Extremely sorry for the misunderstanding. I mixed up two completely two different things. I should have been more alert than before.

Could you please remove the replies related to my issue? That would be really nice of you. I don't want to piss off developers for my ignorance. I appreciate your hard work and I really beg your pardon from the bottom of my heart.

**Comment by nbmrjuhneibkr** [comment]
@avently 
>You don't know how all the code works.
>Notification is the only solution for playing something
>When popup and background players work both notifications appear.

Yes, I don't know how the code works, but I do know how NewPipe behaves now and how it differs from pre-v0.20 behavior. My reference is this - https://github.com/XiangRongLin/NewPipe-preuinified , since it's still maintained and works properly on my device. This version does not display any notifications when a single video is being played in foreground, and this does not affect the background player queue/notification in any way. Also, I never said anything about the _popup_ player, since I have no experience using it.

>I would say that it is possible to have separate queues

As I already said, at its core this issue was never about having _control over multiple queues_, since they are not needed for the usage scenario described in this issue.
The goal should be pretty simple: play individual video without ever adding it to _any_ queue, thus leaving the background queue unaffected by video playback. This is how NewPipe v0.19 worked, and it's one of the main reasons why https://github.com/TeamNewPipe/NewPipe/issues/4918 is pinned.

**Comment by yashpalgoyal1304** [comment]
thanks for sharing Newpipe-preunified's repo. Is that repo updated to contain the latest fixes for playing youtube i.e. does youtube videos play in your device on that app??

**Comment by nbmrjuhneibkr** [comment]
@yashpalgoyal1304 It's NewPipe 0.19.8 + NewPipeExtractor 0.20.11 (and maybe some additional fixes). It is fully functional (at least for now).

**Comment by MapleWheels** [comment]
> @avently
> 
> > You don't know how all the code works.
> > Notification is the only solution for playing something
> > When popup and background players work both notifications appear.
> 
> Yes, I don't know how the code works, but I do know how NewPipe behaves now and how it differs from pre-v0.20 behavior. My reference is this - https://github.com/XiangRongLin/NewPipe-preuinified , since it's still maintained and works properly on my device. This version does not display any notifications when a single video is being played in foreground, and, more importantly, this does not affect the background player queue/notification in any way. Also, I never said anything about the _popup_ player, since I have no experience using it.
> 
> > I would say that it is possible to have separate queues
> 
> As I already said, at its core this issue was never about having _control over multiple queues_, since they are not needed for the usage scenario described in this issue.
> The goal should be pretty simple: play individual video without ever adding it to _any_ queue, thus leaving the background queue unaffected by video playback. This is how NewPipe v0.19 worked, and it's one of the main reasons why #4918 is pinned.

What do you guys think about this solution? https://github.com/TeamNewPipe/NewPipe/issues/5864#issuecomment-808837963

**Comment by SameenAhnaf** [comment]
If I am not wrong, what about this https://github.com/TeamNewPipe/NewPipe/issues/5726? It will avoid messing up queues. Plus, any video or queue could be played right now. They could be managed any time as well.

**Comment by Flashwalker** [comment]
> ... My reference is this - https://github.com/XiangRongLin/NewPipe-preuinified , since it's still maintained and works properly on my device...

Oh, hallelujah! At last someone made that fork!

**Comment by yashpalgoyal1304** [comment]
this is also the side-effect of unified player rework: https://github.com/TeamNewPipe/NewPipe/issues/5304

**Comment by AbakumovAlexandr** [comment]
So, where is a resolution for this issue now at (2 years later)?

I've read multiple related issues and still have no idea why the separate queues for foreground\background players had to be broken unconditionally. This really limited valid use cases of the app a lot.

**Comment by MapleWheels** [comment]
> So, where is a resolution for this issue now at (2 years later)?
> 
> I've read multiple related issues and still have no idea why the separate queues for foreground\background players had to be broken unconditionally. This really limited valid use cases of the app a lot.

This issue is dead. They have no intention of addressing this. I'm just using old new pipe maintained here: https://github.com/XiangRongLin/NewPipe-preuinified along side the new version of new pipe for video content.

Don't waste your breath.

**Comment by AbakumovAlexandr** [comment]
@MapleWheels thanks for the fast reply.

> I'm just using old new pipe maintained here: https://github.com/XiangRongLin/NewPipe-preuinified along side the new version of new pipe for video content.

Yes, I would probably do so as well. Why do you have to use the new version of NewPipe? Does NewPipe-preuinified have some issues even with the back-ported YouTube extractor?

> Don't waste your breath. After going through the runaround on this repo, I think I can finally make the claim that the maintainers are unable to understand the simple, very common use case of wanting to watch a video without affecting the existing queue.

Exactly. I've come to the same conclusion as you just when I read a comment that that dev who made these massive awesome changes was actually a new dev on this project without any overseeing.

Also, I personally just doesn't understand a software development approach when some deal breaking changes on an app with a huge user adoption are being made without keeping any fallback to the previous behavior.

**Comment by cyber199** [comment]
> > So, where is a resolution for this issue now at (2 years later)?
> > I've read multiple related issues and still have no idea why the separate queues for foreground\background players had to be broken unconditionally. This really limited valid use cases of the app a lot.
> 
> This issue is dead. They have no intention of addressing this. I'm just using old new pipe maintained here: https://github.com/XiangRongLin/NewPipe-preuinified along side the new version of new pipe for video content.
> 
> Don't waste your breath. After going through the runaround on this repo, I think I can finally make the claim that the maintainers are unable to understand the simple, very common use case of wanting to watch a video without affecting the existing queue (ie. someone sends me a video over a messenger and I can just pause my music, watch it, and then continue playing the playlist).

I did not know this fork was a thing.  I have been using the old NewPipe Legacy app alongside standard NewPipe just to avoid the very problem of not interrupting my existing queue when someone sends me a video.
